src/Microsoft.ComponentDetection.Detectors/nuget/FrameworkPackages/FrameworkPackages.cs

```diff
@@ -0,0 +1,116 @@
+namespace Microsoft.ComponentDetection.Detectors.NuGet;
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using global::NuGet.Frameworks;
+using global::NuGet.Versioning;
+
+/// <summary>
+/// Represents a set of packages that are provided by a specific framework.
+/// At the moment this only represents the packages that are provided by the Microsoft.NETCore.App framework.
+/// We could extend this to represent the packages provided by other frameworks like Microsoft.AspNetCore.App and Microsoft.WindowsDesktop.App.
+/// </summary>
+internal sealed partial class FrameworkPackages : IEnumerable<KeyValuePair<string, NuGetVersion>>, IEnumerable
+{
+    private static readonly Dictionary<NuGetFramework, FrameworkPackages> FrameworkPackagesByFramework = [];
+
+    static FrameworkPackages()
+    {
+        AddPackages(NETStandard20.Instance);
+        AddPackages(NETStandard21.Instance);
+        AddPackages(NETCoreApp20.Instance);
+        AddPackages(NETCoreApp21.Instance);
+        AddPackages(NETCoreApp30.Instance);
+        AddPackages(NETCoreApp31.Instance);
+        AddPackages(NETCoreApp50.Instance);
+        AddPackages(NETCoreApp60.Instance);
+        AddPackages(NETCoreApp70.Instance);
+        AddPackages(NETCoreApp80.Instance);
+        AddPackages(NETCoreApp90.Instance);
+
+        static void AddPackages(FrameworkPackages packages) => FrameworkPackagesByFramework[packages.Framework] = packages;
+    }
+
+    public FrameworkPackages(NuGetFramework framework) => this.Framework = framework;
+
+    public FrameworkPackages(NuGetFramework framework, FrameworkPackages frameworkPackages)
+        : this(framework) => this.Packages = new(frameworkPackages.Packages);
+
+    public NuGetFramework Framework { get; }
+
+    public Dictionary<string, NuGetVersion> Packages { get; } = new Dictionary<string, NuGetVersion>(StringComparer.OrdinalIgnoreCase);
+
+    public static FrameworkPackages GetFrameworkPackages(NuGetFramework framework)
+    {
+        if (FrameworkPackagesByFramework.TryGetValue(framework, out var frameworkPackages))
+        {
+            return frameworkPackages;
+        }
+
+        // if we didn't predefine the package overrides, load them from the targeting pack
+        // we might just leave this out since in future frameworks we'll have this functionality built into NuGet.
```

Comment:
Any link to public documentation about this coming out later on? You share offline with me if it is a private wiki.

Reply:
Here's the design doc NuGet is working on: https://github.com/NuGet/Home/pull/13634